pkg/util/gvk-gvr-mapper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GvkGvrMapper is a thread-safe mapping between GVKs and GVRs.
```

Comment:
The comments here do not even imagine that there might be a question of whether the relation between GVK and GVR is 1:1. The comments here do not say anything --- exclusive or inclusive --- about subresources.

Reply:
I didn't refer to these cases in this version, but it is doable without affecting the rest of the code since it is interfaced. Do you think this should be updated as part of this PR? or commented as TODO?

Reply:
Not sure what you mean by "all cases". I think updating this one comment is sufficient. Just state that this mapper is only for top-level resources and there is an expectation that, within that restricted scope, the relation between resource and Kind is 1:1.